chapters/en/slides/module5/module5_06.Rmd

```diff
@@ -0,0 +1,288 @@
+
+The default formatting is not very easy to read,
+so let's use what we learned previously to reformat the label
+and make this chart more effective.
+
+---
+
+## Text marks can be formatted in the same manner as axes labels
+
+```{python}
+chart + chart.mark_text(align='left', dx=2).encode(text=alt.Text('sum(sum)', format='$,d'))
+```
+
+Notes:
+`alt.Text` uses the `format` parameter directly instead of within `alt.Axis`.
+Here we add a thousands separator (`,`)
+and round the values to integers (`d`) to remove the decimals.
+
+We also use `dx` to indtroduce some whitespace
+between the bar and the label.
+
+This makes the labels easier to read,
+but since we have thousands of thousands,
+we could simplify further by using the millions suffix.
+
+---
+
+## Standard international units can shorten labels of large values
+
+```{python}
+chart + chart.mark_text(align='left', dx=2).encode(text=alt.Text('sum(sum)', format='$.3~s'))
+```
+
+Notes:
+The `s` format string changes the values to millions
+as per the standard international units convention
+that we discussed earlier.
+
+We use `.3` to indicate that we want three significant digits for each label,
+otherwise they would all be printed with six decimals or so,
+which is hard to read.
+
+We also specify `~`,
+which removes trailing zero,
+e.g. the top bar says `15` instead of `15.0`.
+
+---
+
+## The redundant x-axis can be removed
+
+```{python}
+title = alt.TitleParams(
+    text='Wikipedia receives most donations on Wednesdays',
+    subtitle='Values represent the total amount donated in 2020',
+    anchor='start', dx=26, dy=-5)
+chart = alt.Chart(donations, title=title).mark_bar().encode(
+    alt.Y('week_day', sort=list(day_abbr), title=None),
+    alt.X('sum(sum)', axis=None),
+    color = alt.condition(alt.datum.week_day == 'Wed',
+                          alt.value('coral'),
+                          alt.value('steelblue')))
+(chart + chart.mark_text(align='left', dx=2).encode(text=alt.Text('sum(sum)', format='$.3s'))).configure_view(strokeWidth=0)
+```
+
+Notes:
+We can make our chart stand out by styling it further.
+Here we remove the x-axis,
+which is now redundant
+as we have included its information directly in the text annotations.
+
+We add a title to the plot with the take home message
+and explain what the values are in the subtitle.
+We anchor the title to the start (left) of the plot
+and use `dx` to align it with the bottom of the bars 
+since this often looks more visually appealing for horizontal bar chart.
+
+We use `dy` to add additional whitespace between the subtitle and the topmost bar
+and remove the grey ourline by setting the `strokeWidth` to 0.
+
+---
+
+## Themes can be used to change the overall styling of the chart
+
+```{python}
+alt.themes.enable('dark');
+((chart
+ + chart
+  .mark_text(align='left', dx=2)
+  .encode(text=alt.Text('sum(sum)', format='$.3s')))
+ .configure_view(strokeWidth=0)
+ .configure_title(subtitleColor='white'))
```

Comment:
![image](https://user-images.githubusercontent.com/4560057/108546863-91b6af00-729e-11eb-8ed1-665563c25d5b.png)

Reply:
Is there a nicer theme than this? I thing the black background makes the colours chosen here not that effective...

Reply:
538, latimes or vox might be nice to show?

Reply:
Unfortunately not, they're all a bit odd or don't do much, maybe vox but I also think there is more utility in showing a dark theme since the default is light. You can see all themes here https://vega.github.io/vega-themes/?theme=fivethirtyeight